### PR TITLE
Use latest etcd from release-3.3 branch for dropping ugorji

### PR DESCRIPTION
**Pull request by dims**
Change-Id: I3fb80ac61cfc3b7c549142be2d6c6d36766a58d4

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
etcd has been updated to drop ugorji dependency and switch over to json-iterator just like what we did in the kubernetes repository. Let's pick that change up so we can get rid of a bunch of code.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: vendor/github.com/coreos/etcd/client/keys.go [hunk 1]
```diff
@@ -658,7 +656,7 @@ func unmarshalHTTPResponse(code int, header http.Header, body []byte) (res *Resp
 
 func unmarshalSuccessfulKeysResponse(header http.Header, body []byte) (*Response, error) {
 	var res Response
-	err := codec.NewDecoderBytes(body, new(codec.JsonHandle)).Decode(&res)
+	err := jsoniter.ConfigFastest.Unmarshal(body, &res)
```
Comment (liggitt) [inline vendor/github.com/coreos/etcd/client/keys.go]: need to use stdlib-compatible one that is case-sensitive, etc

**Comment by liggitt** [comment]
I actually think it would be better to pursue eliminating use of the ".../coreos/etcd/client" package. We shouldn't be using the v2 client for anything at this point, and we should request the constants we're referencing be copied to the v3 package

**Comment by dims** [comment]
@liggitt unfortunately that won't be enough as we indirectly pull in ugorji stuff as ".../coreos/etcd/client" is used in ".../coreos/etcd/integration" and ".../coreos/etcd/discovery"

File: vendor/github.com/coreos/etcd/client/keys.go (hunk 1 above)
Comment (dims) [reply vendor/github.com/coreos/etcd/client/keys.go]: ack switching to `jsoniter.ConfigCompatibleWithStandardLibrary`

Comment (liggitt) [reply vendor/github.com/coreos/etcd/client/keys.go]: actually, ConfigCompatibleWithStandardLibrary is not case-sensitive, and we *really* want it to be (c.f. https://github.com/kubernetes/kubernetes/pull/65034, https://github.com/kubernetes/kubernetes/issues/64612)

Comment (liggitt) [reply vendor/github.com/coreos/etcd/client/keys.go]: for comparison, ugorji is case-sensitive

Comment (dims) [reply vendor/github.com/coreos/etcd/client/keys.go]: Ack @liggitt Let's do this in https://github.com/etcd-io/etcd/pull/10667/commits and then bring back the changes here for testing.

**Comment by liggitt** [comment]
setting this in the `replace` portion of `go.mod` and running hack/update-vendor.sh would let you try this with exactly your branch:

```
github.com/coreos/etcd => github.com/dims/etcd v0.0.0-20190423213535-99140863f9cc
```

**Comment by dims** [comment]
size - savings 

```
dims@bigbox:~/junk$ diff -Bbwu before.raw after.raw
--- before.raw	2019-04-25 17:06:56.865091360 -0400
+++ after.raw	2019-04-25 17:03:53.663533820 -0400
@@ -490,6 +490,7 @@
 528K	./k8s.io/kubernetes/vendor/github.com/Azure/go-autorest/autorest.a
 544K	./k8s.io/kubernetes/vendor/github.com/google/cadvisor/manager.a
 556K	./k8s.io/kubernetes/vendor/github.com/libopenstorage/openstorage/api.a
+560K	./k8s.io/kubernetes/vendor/github.com/coreos/etcd/client.a
 564K	./k8s.io/kubernetes/vendor/github.com/asaskevich/govalidator.a
 592K	./k8s.io/kubernetes/vendor/gopkg.in/square/go-jose.v2/json.a
 600K	./k8s.io/kubernetes/vendor/google.golang.org/api/tpu/v1.a
@@ -511,7 +512,6 @@
 792K	./k8s.io/kubernetes/vendor/github.com/gogo/protobuf/protoc-gen-gogo/descriptor.a
 812K	./k8s.io/kubernetes/vendor/gopkg.in/square/go-jose.v2.a
 896K	./k8s.io/kubernetes/vendor/github.com/pkg/sftp.a
-908K	./k8s.io/kubernetes/vendor/github.com/coreos/etcd/client.a
 956K	./k8s.io/kubernetes/vendor/github.com/gogo/protobuf/types.a
 964K	./k8s.io/kubernetes/vendor/github.com/go-openapi/spec.a
 968K	./k8s.io/kubernetes/vendor/github.com/containerd/containerd/api/services/tasks/v1.a
@@ -557,7 +557,6 @@
 4.4M	./k8s.io/kubernetes/vendor/github.com/rancher/go-rancher/client.a
 6.3M	./k8s.io/kubernetes/vendor/github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud.a
 8.5M	./k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute.a
-9.7M	./k8s.io/kubernetes/vendor/github.com/ugorji/go/codec.a
 14M	./k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network.a
 14M	./k8s.io/kubernetes/vendor/google.golang.org/api/compute/v1.a
 15M	./k8s.io/kubernetes/vendor/github.com/xanzy/go-cloudstack/cloudstack.a
```

**Comment by dims** [comment]
size of hyperkube
before
```
-rwxrwxr-x 1 dims dims 193M Apr 25 17:10 _output/local/go/bin/hyperkube
```

after
```
-rwxrwxr-x 1 dims dims 190M Apr 25 17:12 _output/local/bin/linux/amd64/hyperkube
```

File: go.mod [hunk 2]
```diff
@@ -37,7 +37,7 @@ require (
 	github.com/containerd/containerd v1.0.2 // indirect
 	github.com/containerd/typeurl v0.0.0-20190228175220-2a93cfde8c20 // indirect
 	github.com/containernetworking/cni v0.6.0
-	github.com/coreos/etcd v3.3.10+incompatible
+	github.com/coreos/etcd v3.3.13-0.20190425194549-ad7c2cddb000+incompatible
```
Comment (liggitt) [inline go.mod]: would prefer not to move off a tagged version for an untagged one. anything in flight in the 3.3.x stream we should be concerned about picking up half-baked?

Comment (dims) [reply go.mod]: @liggitt how about i ask for a new release just to be sure :)

**Comment by dims** [comment]
fyi, looks like we may see a 3.4 for etcd within a week. 

/hold

**Comment by dims** [comment]
@liggitt updated using `hack/pin-dependency.sh github.com/coreos/etcd 98d308426819d892e149fe45f6fd542464cb1f9d` 

3.3.14 tag was cut today at that SHA https://github.com/etcd-io/etcd/tree/v3.3.13

/hold cancel

Thanks!

**Pull request merged**